### PR TITLE
Add command line args for example scripts

### PR DESCRIPTION
**Pull request by mschmidt87**
This PR implements command-line args for the example scripts. It is the first step to address #208 . In a follow-up PR, I will then make use of these command-line args in the travis CI.

The following things are implemented:
- Add option to specify max_generations on the command line to example scripts
- Add switch for visualization in mountain_car example

### CODE REVIEW

File: examples/example_differential_evo_regression.py [hunk 1]
```diff
@@ -28,6 +29,16 @@
 
 import cgp
 
+# %%
```
Comment (jakobj) [inline examples/example_differential_evo_regression.py:32]: mandatory comment on [docopt](http://docopt.org/) being much cooler than argparse ;)

File: examples/example_mountain_car.py [hunk 2]
```diff
@@ -29,6 +29,26 @@
...
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--max-generations", type=int, help="max. number of generations in the evolution", default=1500
+)
+visualize_parser = parser.add_mutually_exclusive_group(required=False)
+visualize_parser.add_argument(
+    "--visualize-final-champion", dest="visualize_final_champion", action="store_true"
+)
+visualize_parser.add_argument(
```
Comment (jakobj) [inline examples/example_mountain_car.py:46]: i don't understand why you need this option. isn't it the case that if `--visualize-final-champion` is provided it's true, if it's not provided it's false?

Comment (mschmidt87) [reply examples/example_mountain_car.py]: solved with replacing by `docopt`

File: examples/example_differential_evo_regression.py (hunk 1 above)
Comment (mschmidt87) [reply examples/example_differential_evo_regression.py]: Yes, I agree, `docopt` is much nicer and shorter. However, strangely, `docopt` itself is basically not maintained anymore. See this issue: https://github.com/docopt/docopt/issues/371 with some strange replies by the maintainer (?) who "just doesn't have the time to reject all the PRs" ...

Therefore I would prefer to use [`docopt-ng`](https://github.com/bazaar-projects/docopt-ng), a maintained fork.

Two things are suboptimal in my opinion:
1. docopt does not infer the types of arguments? That is why I need to explicitly convert the value from `str` to `int`. Seems strange that they don't offer that.
2. They also just take over the argument as is, i.e. `--max_generations` is not turned into `max_generations` (what `argparse` is offering). That is also a bit strange.

Perhaps I am just not understanding how to it, though. Do you have an idea?

Comment (jakobj) [reply examples/example_differential_evo_regression.py]: > Two things are suboptimal in my opinion:
> 
>     1. docopt does not infer the types of arguments? That is why I need to explicitly convert the value from `str` to `int`. Seems strange that they don't offer that.
> 
>     2. They also just take over the argument as is, i.e. `--max_generations` is not turned into `max_generations` (what `argparse` is offering). That is also a bit strange.
> 
> 
> Perhaps I am just not understanding how to it, though. Do you have an idea?

i think you are right in both cases. you need to manually convert the arguments from str and they keep their name from the docstring. i would argue that both are not an issue tough.

File: examples/example_evo_regression.py [hunk 3]
```diff
@@ -15,6 +23,8 @@
 
 import cgp
 
+args = docopt(__doc__, more_magic=True)
+breakpoint()
```
Comment (jakobj) [inline examples/example_evo_regression.py:27]: pls remove

File: examples/example_differential_evo_regression.py [hunk 4]
```diff
@@ -18,8 +18,16 @@
   programming. In European Conference on Genetic Programming
   (pp. 35-51). Springer, Cham.
 
+Usage: 
+  example_differential_evo_regression.py [--max_generations=<N>]
```
Comment (jakobj) [inline examples/example_differential_evo_regression.py:22]: i would prefer command line argument to be dash-separated like this `--max-generations`

File: examples/example_evo_regression.py (hunk 3 above)
Comment (mschmidt87) [reply examples/example_evo_regression.py:27]: thanks, done.

File: examples/example_differential_evo_regression.py (hunk 4 above)
Comment (mschmidt87) [reply examples/example_differential_evo_regression.py]: done.

File: examples/example_differential_evo_regression.py (hunk 1 above)
Comment (mschmidt87) [reply examples/example_differential_evo_regression.py]: Yeah, not an issue, but quite ugly. I also don't understand, why docopt doesn't infer the type from the default value (or optional documentation).

Comment (mschmidt87) [reply examples/example_differential_evo_regression.py]: This used to be the case, but was then silently removed: https://www.google.com/search?hl=en&q=docopt%20infer%20type

Comment (mschmidt87) [reply examples/example_differential_evo_regression.py]: Anyway, it is what it is.

**Comment by mschmidt87** [comment]
I resolved the issues and already squashed. Please take a look @jakobj

**Comment by mschmidt87** [comment]
Turns out that using `docopt` creates problems with sphinx-gallery because sphinx-gallery "steals" the first docstring of the scripts to render it in html and then docopt can no longer consume it to create the CLI. (see https://github.com/sphinx-gallery/sphinx-gallery/issues/341 for related discussion). 
Therefore I refactored the code slightly. It is not so nice anymore, but I think this it the only way to make `docopt` work with `sphinx-gallery`.

**Pull request merged**